app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java

```diff
@@ -103,25 +115,14 @@ public void onCreate() {
         Toast.makeText(this, "using db: "+getDataPath(), Toast.LENGTH_SHORT);
 
         mTelephonyManager = (TelephonyManager) getSystemService(Context.TELEPHONY_SERVICE);
+        mTimer = new Timer();
 
-        // schedule a periodic update
-        final Handler handler = new Handler();
-        Runnable timer = new Runnable() {
-            @Override
-            public void run() {
-                if (running) {
-                    updateCellInfo();
-                    handler.postDelayed(this, App.UPDATE_DELAY_MILLIS);
-                }
-            }
-        };
-        handler.post(timer);
     }
 
     @Override
     public void onDestroy() {
         Log.v(App.TITLE, getClass().getName()+".onDestroy()");
-
+        mTimer.cancel();
```

Comment:
Timer is started on start command. Matching stop trigger would be on stop command

Reply:
Service::onStopCommand | Service:onStop is non-existend.

'onDestroy()
    The system invokes this method when the service is no longer used and is being destroyed. Your service should implement this to clean up any resources such as threads, registered listeners, or receivers. This is the last call that the service receives.

https://developer.android.com/guide/components/services
'